suite-native/module-send-receive/src/components/FreshAddressStep.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useEffect, useState } from 'react';
+import { Alert, Share, View } from 'react-native';
+import QRCode from 'react-qr-code';
+import { useSelector } from 'react-redux';
+
+import Clipboard from '@react-native-clipboard/clipboard';
+
+import { Box, Button, HStack, Text } from '@suite-native/atoms';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import { defaultColorVariant } from '@trezor/theme';
+import { selectAccountTransactions, TransactionsRootState } from '@suite-common/wallet-core';
+import { AccountAddress } from '@trezor/connect';
+import { getFirstFreshAddress, getPendingAddresses, isUtxoBased } from '@suite-common/wallet-utils';
+import { Account } from '@suite-common/wallet-types';
+
+type FreshAddressStepProps = {
+    account: Account;
+    onClose: () => void;
+};
+
+export const QRCODE_SIZE = 197;
+export const QRCODE_PADDING = 12;
+
+const qrCodeWrapperStyle = prepareNativeStyle(_ => ({
+    padding: QRCODE_PADDING,
+    alignItems: 'center',
+    justifyContent: 'center',
+}));
+
+const actionButtonsStyle = prepareNativeStyle(_ => ({
+    alignItems: 'center',
+    justifyContent: 'center',
+    flexDirection: 'row',
+}));
+
+export const FreshAddressStep = ({ account, onClose }: FreshAddressStepProps) => {
+    const { applyStyle } = useNativeStyles();
+    const [freshAddress, setFreshAddress] = useState<AccountAddress>();
+
+    const accountTransactions = useSelector((state: TransactionsRootState) =>
+        selectAccountTransactions(state, account?.key),
+    );
+
+    useEffect(() => {
+        if (account && accountTransactions) {
+            const pendingAddresses = getPendingAddresses(accountTransactions);
+            const utxoBasedAccount = isUtxoBased(account);
+            // Pass receive addresses empty. We proceed without address verification - not possible in watch only mode.
+            const firstFreshAddress = getFirstFreshAddress(
```

Comment:
I created two selectors for getting pending addresses and to check if an account is utxo based. But... I would rather not create a selector for getting a fresh address as you proposed here because `getFirstFreshAddress` accepts parameters from 3 reducers (transactions, receive, accounts) and I am worried it would affect readability. So I have used the solution with useMemo c0a5797713ec5624f366feaa53bcc530f13ce107